src/test/ui/numeric/numeric-cast-no-fix.rs

```diff
@@ -0,0 +1,22 @@
+#[allow(unused_must_use)]
+fn main() {
+    let x_usize: usize = 1;
+    let x_u128: u128 = 2;
+    let x_u64: u64 = 3;
+    let x_u32: u32 = 4;
+    let x_u16: u16 = 5;
+    let x_u8: u8 = 6;
+
+    x_usize > -1_isize;
+    //~^ ERROR mismatched types
+    x_u128 > -1_isize;
+    //~^ ERROR mismatched types
+    x_u64 > -1_isize;
+    //~^ ERROR mismatched types
+    x_u32 > -1_isize;
+    //~^ ERROR mismatched types
+    x_u16 > -1_isize;
+    //~^ ERROR mismatched types
+    x_u8 > -1_isize;
+    //~^ ERROR mismatched types
```

Comment:
Can we do this exhaustive by including the `i*` cases as well? A bunch of those should also suggest `<type>::from` like the `u8` case does. I trust that they do (given the code above), but we also want to catch regressions.

Reply:
Also, let's add a check for `f32`/`f64` as well, while we are at it (to make sure we don't suggest the wrong thing).